File: app/src/main/java/com/medsdate/data/db/MedicineDao.java
package com.medsdate.data.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.medsdate.data.db.model.MedicineEntry;

import java.util.List;

@Dao
public interface MedicineDao {

    @Query("SELECT * FROM medicine ORDER BY expire_at ASC")
    LiveData<List<MedicineEntry>> loadAllMeds();

    @Insert
    void insertMedicine(MedicineEntry medicineEntry);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateMedicine(MedicineEntry medicineEntry);

    @Delete
    void deleteMedicine(MedicineEntry medicineEntry);

    @Query("SELECT * FROM medicine WHERE id = :id")
    LiveData<MedicineEntry> loadMedicineById(int id);
}

File: app/src/main/java/com/medsdate/data/db/MedicineDao.kt
package com.medsdate.data.db

import androidx.lifecycle.LiveData
import androidx.room.*
import com.medsdate.data.db.model.MedicineEntry

@Dao
interface MedicineDao {

    @Query("SELECT * FROM medicine ORDER BY expire_at ASC")
    fun loadAllMeds(): LiveData<List<MedicineEntry>>

    @Insert
    fun insertMedicine(medicineEntry: MedicineEntry)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateMedicine(medicineEntry: MedicineEntry)

    @Delete
    fun deleteMedicine(medicineEntry: MedicineEntry)

    @Query("SELECT * FROM medicine WHERE id = :id")
    fun loadMedicineById(id: Int): LiveData<MedicineEntry>
}
